src/main.ts

```diff
@@ -553,6 +557,71 @@ class SettingTab extends PluginSettingTab {
           });
         });
   }
+
+  addCustomCommandsSetting(): void {
+    this.containerEl.createEl('h2', {text: 'Custom Commands Settings'});
+    function arrayMove(arr:string[], fromIndex:number, toIndex:number):void {
+      if (toIndex < 0 || toIndex === arr.length) {
+        return;
+      }
+      const element = arr[fromIndex];
+      arr[fromIndex] = arr[toIndex];
+      arr[toIndex] = element;
+    }
+    new Setting(this.containerEl)
+        .addButton((cb)=>{
+          cb.setButtonText('Add new command')
+              .setCta()
+              .onClick(()=>{
+                this.plugin.settings.lintCommands.push('');
```

Comment:
You seem to be pushing an empty value instead of the selected value, but maybe that was something you planned to add yet.

Reply:
Yes, this initializes an empty string when the command is added, and is later changed by the user.

Reply:
Interesting. It might help to add a comment saying that is the case to make it clear what is happening as I guess I just misunderstood it.